src/main/java/com/javagirls/MangoRoom/service/ReservationService.java

```diff
@@ -94,4 +96,25 @@ public void changeReservationStatus(Long id, Status status) {
         return reservationRepository.findAll().stream()
                 .map((reservation) -> mapper.map(reservation, ReservationDto.class)).collect(Collectors.toList());
     }
+
+    public List<RoomDto> getFreeRooms(LocalDateTime startDate, LocalDateTime endDate){
+        List<ReservationDto> reserved = new ArrayList<>();
```

Comment:
1. cały ten stream przypisałabym od razu do listy reserved, wtedy nie musisz dodawać każdego elementy do nowej listy, tylko filtruje z tego co dostał w findAll
2. sprawdź czy działa za pomocą Swaggera :)

Reply:
nie wiem, jak to zrobić :/ nie mam pojęcia o tym streamie, przerobiłam kod, który juz był...